test/structuredbroadcast.jl

```diff
@@ -142,13 +144,15 @@ end
     @test broadcast!(sin, copy(T), T) == Tridiagonal(sin.(T))
     @test broadcast!(sin, copy(◣), ◣) == LowerTriangular(sin.(◣))
     @test broadcast!(sin, copy(◥), ◥) == UpperTriangular(sin.(◥))
+    @test broadcast!(sin, copy(UH), UH) == UpperHessenberg(sin.(UH))
```

Comment:
```suggestion
    @test broadcast!(sin, copy(UH), UH)::UpperHessenberg == UpperHessenberg(sin.(UH)::UpperHessenberg)
```
Add some type asserts to make sure it retains the correct type?